lib/storage/src/content_manager/toc.rs

```diff
@@ -997,6 +1023,18 @@ impl TableOfContent {
         false
     }
 
+    pub fn set_write_lock(&self, locked: bool, error_message: Option<String>) -> bool {
+        let prev_value = self
+            .write_lock
+            .swap(locked, std::sync::atomic::Ordering::Relaxed);
```

Comment:
nitpick: it would nice to capture the date at which the lock was enabled.

This would enable to have a `since` field in the error message by default.

Reply:
IMHO Not needed. Because this `since` cloud team can format itself in the error message. Only they know about the reasons for blocking and only they should decide about `since` timestamp